djangocms_page_sitemap/sitemap.py

```diff
@@ -14,10 +19,44 @@ class ExtendedSitemap(CMSSitemap):
     default_priority = CMSSitemap.priority
 
     def items(self):
-        return super(ExtendedSitemap, self).items().exclude(
-            page__pagesitemapproperties__include_in_sitemap=False
+        # This method was created from this commit:
+        # https://github.com/divio/django-cms/blob/2894ae8bcf92092d947a097499c01ab2bbb0e6df/cms/sitemaps/cms_sitemap.py
```

Comment:
Wouldn't be a better approach on the upstream `items` to do something like

```python
        ...
        excluded_pks = []
        for page_url in all_urls:
            excluded = excluded_titles_by_page.get(page_url.page_id, [])

            if page_url.language in excluded:
                excluded_pks.append(page_url.pk)
        # select related can be removed from the first all_url queryset
        return all_urls.exclude(pk__in=excluded_pks).select_related('page')
```
that will return a queryset and allow cleaner manipulation when extending the sitemap

the impact of the additional query should be low enough to not be a concern

I am afraid that the current implementation of `ExtendedSitemap.items` as proposed here is very brittle and will require a lot of maintenance

Reply:
@yakky You have the freedom to change the implementation here, I wouldn't request a change on the core when this package copies the logic anyway, if you need to change it to suit your package you have the freedom to do so here.

Reply:
@yakky I don't think that your requested implmentation satisfies the required addition from this package: `                page__pagesitemapproperties__include_in_sitemap=False`

Reply:
@yakky I would like to add that you now actually have more freedom to extend the implementation further in the future.

Reply:
@yakky Do you want us to add the implementation that you have specified here instead of the core implementation? I have looked at this for a while now and I can see that the items used to return the Title object, it's now returning a list instead of expecting a Title Queryset. I personally believe that the list gives full flexibility to third party packages allowing developers to completely redesign the functionality. 

As mentioned, if you would like us to rewrite this implementation here to what you have created please let us know.

Reply:
I do appreciate it would make this package cleaner, I also see the other scenarios of the fact that the core by default performs faster and developers have complete flexibility of changing how the items are generated by not forcing it to  model queryset.

Reply:
the problem is that the code here is a copy/paste of the django CMS code, putting a lot of maintenance burden on this app for whatever change happens in the CMS
an approach which would allow `items` to return a queryset would solve any extensibility issue (and it would cleaner in the first place)

Reply:
@yakky There has been a decision to no longer fix items to returning a queryset, I believe to allow flexibility by returning a list rather than tying the returned data to a particular queryset, I think that your request then undoes that change and adds a performance hit by making an additional query to reattach pages. We don't have to do a copy and paste here if you don't want to, we can reshape it to better fit the needs of this package which is why I also believe that the implementation was changed to not return a queryset. You would expect items to return a list by pattern and it was originally an antipattern for "items" to return a queryset. "items" by definition should return a result as a list of items as the name suggests.

Reply:
My concern here is that copy pasting code from django CMS will make this package harder to maintain, any solution which would make this code clean and non duplicate with django CMS will work for me.
WRT return type, I don't see that way (see the examples in the django documentation), but it's not a problem in this context and not a blocker

Reply:
@yakky We will struggle to push this change into the Core right now, as mentioned the move to return items as a list is to allow more control. The comment for the commit that the code was taken from will help to track changes in the future and removing it will make this harder.

Reply:
Please add a FIXME comment